terraform-modules/k8s/k8s-cluster.tf

```diff
@@ -59,24 +56,50 @@ resource "google_container_cluster" "cluster" {
   # CIS compliance: Enable Alias IP Ranges. According to the terrafform
   # docs, setting these values blank gets default-size ranges automatically
   # chosen: https://www.terraform.io/docs/providers/google/r/container_cluster.html#ip_allocation_policy
-  ip_allocation_policy = ["${var.ip_allocation_policy}"]
+  dynamic "ip_allocation_policy" {
```

Comment:
wow this is interesting. Is the idea that the `content` block gets rendered into basically an array of maps?

Reply:
I think `dynamic` ends up rendering a top-level block per element in the iterator, so we end up with something like:
```hcl
ip_allocation_policy {
  cluster_ipv4_cidr_block = "value from object 1"
  # Rest of the fields
}
ip_allocation_policy {
  cluster_ipv4_cidr_block = "value from object 2"
  # Rest of the fields
}
```
I don't know enough about the underlying GKE config to know if it's valid to have multiple policies, or if this was just a way to generically allow users to disable `ip_allocation_policy` by passing an empty list.

Reply:
`terraform init` also doens't complain if I do this:
```hcl
ip_allocation_policy = var.ip_allocation_policy
```
... which is much simpler, but who knows if it means the same thing.

I'm going to test `terraform plan` on my team's concrete deploy using this new module, and see what happens.